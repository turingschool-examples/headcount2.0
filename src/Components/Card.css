.card {
  background-color: #375053;
  border-radius: 15px;
  box-shadow: 0px 0px 19px 1px rgba(0, 0, 0, .7);
  flex: 1 18em;
  list-style-type: none;
  margin: 0 2em 2em 2em;
  padding: 0 1em 1em 1em;
}

.selected {
  background-color: #5e7377;
  border: solid 4px #f7b488;
}

h2 {
  font-size: 2em;
  margin: .5em;
  text-align: center;
}

li {
  font-size: 1.2em;
  padding: 0 0 0 6em;
}

.stat {
  padding: 0 0 .5em 0;
}

.less-than {
  color: #f7b488;
}