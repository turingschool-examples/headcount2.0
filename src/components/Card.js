import React from 'react';
import PropTypes from 'prop-types';


const Card = ({ location, data, handleCompare, compare, active }) => {
  const classname = active === true ? 'card active' : 'card'
  return (
    <div className={classname} key={location} onClick={() => handleCompare(location)}>
      <h1 className='cardTitle'>
        {location}
      </h1>
      {Object.keys(data).map((key, index) => {


        return (
          <div className='cardData' key={index}>
            <li className={
              data[key] <= 0.5 ? 'negative' : 'positive'
            }> {key}: {data[key]} </li>
          </div>
        )
      })}
    </div>
  );
}

Card.propTypes = {
  location: PropTypes.string,
  data: PropTypes.object,
  handleCompare: PropTypes.func,
}

export default Card;
