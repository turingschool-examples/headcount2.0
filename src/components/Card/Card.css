/*CARD COMPONENT RELATED CSS GOES HERE*/
.Card {
  height: 350px;
  width: 250px;
  margin: 5px 0;
  padding: 0.5rem;
  padding-bottom: 1.25rem;
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: 4rem 1fr;
  justify-items: center;
  color: #006989;
  letter-spacing: 0.5px;
  border-radius: 5%;
  background: rgb(255, 255, 255);
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
  box-sizing: border-box;
}

.Card h2 {
  text-align: center;
  grid-row: 1 / 2;
  font-family: "Roboto Condensed", sans-serif;
  font-weight: 700;
  font-size: 20px;
  margin: 0;
  padding-top: 1.25rem;
}

ul {
  grid-row: 2 / 3;
  list-style-type: none;
  font-family: "Roboto", sans-serif;
  font-weight: 400;
  text-size: 1.5rem;
  padding-left: 0;
  align-self: center;
}

.low {
  color: #EF713B;
}

.selected {
  border: 5px solid #EF713B;
}