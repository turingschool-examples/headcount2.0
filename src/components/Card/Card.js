import React from 'react';
import PropTypes from 'prop-types';

const Card = props => {
  const { location, stats } = props;

  const districtStats = Object.keys(stats).map(stat => <li key={stat}>{stats[stat]}</li>);

  return (
    <div>
      <h2>{location}</h2>
      <ul>
        {districtStats}
      </ul>
    </div>
  );
};

Card.propTypes = {
  location: PropTypes.string.isRequired,
  stats: PropTypes.object.isRequired
};

export default Card;
