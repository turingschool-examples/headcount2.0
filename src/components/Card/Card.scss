$backgroundColor: rgba(255, 255, 255, .1);
$sublimeGreen: #AAD74F;
$sublimeRed: #D23D54;
$sublimeBlue: #78C9D0;
$sublimeYellow: #FFFF74;

article {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-bottom: 20px;
  width: 250px;
  border: 1px solid #fff;
  background: $backgroundColor;
}

article h3 {
  color: $sublimeGreen;
  font-size: 1.5em;
  margin-bottom: 0px;
}

article p {
  margin: 7px;
}

ul {
  text-align: left;
  list-style: none;
  padding: 0;

  li {
    margin: 3px 3px;
  }

  .over {
    color: $sublimeBlue;
  }

  .under {
    color: $sublimeRed;
  }

  .fa {
    margin-right: 8px;
  } 

  .fa-bolt {
    color: yellow;
  }

  .fa-tint {
    color: #849FED;
  }
}

.clicked {
  border: 2px solid $sublimeYellow;
}