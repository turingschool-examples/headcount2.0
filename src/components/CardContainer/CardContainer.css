/*CARDCONTAINER COMPONENT RELATED CSS GOES HERE*/
.CardContainer {
  grid-row: 3 / 4;
  grid-column: 1 / 2;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  padding-top: 15px;
}