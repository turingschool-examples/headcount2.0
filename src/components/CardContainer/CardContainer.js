import React from 'react';
import PropTypes from 'prop-types';
import Card from '../Card/Card';

import './CardContainer.css';

const CardContainer = props => {
  const { districts } = props;

  const districtData = districts.map(district => <Card {...district} key={district.location} />);

  return (
    <div>
      <h1>Card Container</h1>
      <div className="card-container">{districtData}</div>
    </div>
  );
};

CardContainer.propTypes = {
  districts: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default CardContainer;
