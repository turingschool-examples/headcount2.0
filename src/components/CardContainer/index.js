import React from 'react'
import PropTypes from 'prop-types'
import Card from '../Card'
import './CardContainer.css'

const CardContainer = ({districts, adjustComparisons}) => {
  const cards = Object.keys(districts).map(district => {
    const stats = districts[district].stats

    return <Card key={`${district}-district`}
                 location={district}
                 stats={stats}
                 adjustComparisons={adjustComparisons}
           />
  })

  return(
    <main className="CardContainer">
       { cards }
    </main>
  )
}

CardContainer.propTypes = {
  districts: PropTypes.object.isRequired
}

export default CardContainer