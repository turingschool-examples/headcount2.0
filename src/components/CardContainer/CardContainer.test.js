import React from 'react'
import { shallow } from 'enzyme'
import CardContainer from './index'
import Card from '../Card'

describe('CardContainer', () => {
  let wrapper;
  let districts;

  beforeEach(() => {
    districts = { 'COLORADO': {}, 'ALAMOSA': {} }
    wrapper = shallow(<CardContainer districts={districts} />)
  })

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should have all the cards', () => {
    expect(wrapper.find(Card).length).toEqual(2)
  })
})