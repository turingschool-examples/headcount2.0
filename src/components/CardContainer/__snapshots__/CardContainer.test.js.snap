// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer matches the snapshot 1`] = `
<main
  className="CardContainer"
>
  <Card
    key="COLORADO-district"
    location="COLORADO"
  />
  <Card
    key="ALAMOSA-district"
    location="ALAMOSA"
  />
</main>
`;
